src/merger.ts

```diff
@@ -210,32 +216,70 @@ export class CodeModelMerger {
             if (!isNullOrUndefined(flattenedNodes) && flattenedNodes.length > 0) {
                 for (const fnode of flattenedNodes) {
                     if (!isNullOrUndefined(fnode) && !isNullOrUndefined(fnode.language)) {
-                        for (const prop of values(param.schema.properties)) {
-                            if (
-                                !isNullOrUndefined(fnode.language?.cli?.cliKey) &&
-                                fnode.language?.cli?.cliKey === prop['language']?.cli?.cliKey
-                            ) {
-                                fnode.language.python = prop['language'].python;
-                                fnode.language.cli.pythonFlattenedFrom = param;
-                                break;
-                            } else if (!isNullOrUndefined(fnode.language?.cli?.cliFlattenTrace)) {
-                                for (const trace of values(fnode.language.cli.cliFlattenTrace)) {
-                                    if (
-                                        !isNullOrUndefined(prop['language']?.cli?.cliPath) &&
-                                        trace === prop['language'].cli.cliPath
-                                    ) {
-                                        for (const p of prop['schema']?.properties) {
-                                            if (
-                                                !isNullOrUndefined(p.language?.cli?.cliKey) &&
-                                                fnode.language?.cli?.cliKey ===
-                                                    p.language?.cli?.cliKey
-                                            ) {
-                                                fnode.language.python = p.language.python;
-                                                fnode.language.cli.pythonFlattenedFrom = prop;
-                                                break;
+                        let foundProp = false;
```

Comment:
In general, i don't suggest writing that complicated logic in one function which is hard to understand and maintain.

Reply:
Is it possible to split into small functions?  If not, could you add some comments about the steps?

Reply:
Added comments